elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py

```diff
@@ -0,0 +1,349 @@
+import json
+from collections import defaultdict
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union
+
+from alive_progress import alive_it
+
+from elementary.config.config import Config
+from elementary.monitor.alerts.group_of_alerts import GroupedByTableAlerts, GroupingType
+from elementary.monitor.alerts.model_alert import ModelAlertModel
+from elementary.monitor.alerts.source_freshness_alert import SourceFreshnessAlertModel
+from elementary.monitor.alerts.test_alert import TestAlertModel
+from elementary.monitor.api.alerts.alerts import AlertsAPI
+from elementary.monitor.api.alerts.schema import AlertsSchema
+from elementary.monitor.data_monitoring.alerts.integrations.slack.slack import (
+    SlackIntegration,
+)
+from elementary.monitor.data_monitoring.data_monitoring import DataMonitoring
+from elementary.monitor.data_monitoring.schema import ALERT_TABLES
+from elementary.monitor.fetchers.alerts.schema import (
+    PendingModelAlertSchema,
+    PendingSourceFreshnessAlertSchema,
+    PendingTestAlertSchema,
+)
+from elementary.tracking.tracking_interface import Tracking
+from elementary.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class DataMonitoringAlerts(DataMonitoring):
+    def __init__(
+        self,
+        config: Config,
+        tracking: Optional[Tracking] = None,
+        filter: Optional[str] = None,
+        force_update_dbt_package: bool = False,
+        disable_samples: bool = False,
+        send_test_message_on_success: bool = False,
+        global_suppression_interval: int = 0,
+        override_config: bool = False,
+    ):
+        super().__init__(
+            config, tracking, force_update_dbt_package, disable_samples, filter
+        )
+
+        self.global_suppression_interval = global_suppression_interval
+        self.override_config = override_config
+        self.alerts_api = AlertsAPI(
+            self.internal_dbt_runner,
+            self.config,
+            self.elementary_database_and_schema,
+            self.global_suppression_interval,
+            self.override_config,
+        )
+        self.sent_alert_count = 0
+        self.send_test_message_on_success = send_test_message_on_success
+        self.override_meta_slack_channel = override_config
+        self.alerts_integraion = self._get_integration_client()
```

Comment:
This is the function that determines which integration we should use.
It can't be part of the integration itself, and once we will add more integrations it should be more dynamic (currently it just returns SlackIntegration)

Reply:
We can create a method / class (outside of the BaseIntegration) that determines which integration we should use (base on params / config) and we can call it at the data monitoring.
We will still have a method that calls it inside the data monitoring but all the logic won't be there.